### PR TITLE
mnw2 export, modpath example and shapefile export, epsg and geometry utils

### PR DESCRIPTION
**Pull request by aleaf**
### MNW2  
* fixed bug with k, i, j node input option
* mnw2 works with netcdf export  

### Shapefile features   
 (see examples/Notebooks/flopy3_shapefile_features.ipynb)
* added generic functions that read and write shapefiles to/from recarrays. This should make it easy to add sparse export (only the cells of interest) to any BC inputs/outputs
* added geometry classes for easier manipulation of vector data with attributes
* added epsg utility for retrieving, storing, and writing prj files using epsg codes
* added epsg and length_multiplier arguments to ```SpatialReference``` class  

### Modpath features  
(see examples/Notebooks/flopy3_Modpath_example.ipynb)
* added ```StartingLocations``` class to mpsim.py (probably still needs some work; had trouble getting MODPATH6 to always work with starting locations file(s))
* added easy subsetting of endpoint and pathline results to destination cells of interest
* added shapefile export of endpoints and pathlines
* for now, in lieu of a refactoring of the ModpathSim class, added MNW2 to ```create_mpsim()``` helper

### CODE REVIEW

**Comment by aleaf** [comment]
also added the MODPATH6 example files to ```examples/data/mp6```

**Comment by aleaf** [comment]
are we still supporting Python 3.2? The main readme says 3.3 or greater. 3.2 was the only build that failed.

File: flopy/modpath/mpsim.py [hunk 1]
```diff
@@ -280,3 +280,94 @@ def write_file(self):
...
+        with open(self.fn_path, 'w') as output:
+            output.write('{}\n'.format(self.heading))
+            output.write('{:d}\n'.format(self.input_style))
+            groups = np.unique(self.data.groupname)
+            ngroups = len(groups)
+            output.write('{:d}\n'.format(ngroups))
+            for g in groups:
+                npt = len(self.data[self.data.groupname == g])
+                output.write('{}\n{:d}\n'.format(g.decode(), npt))
+            for p in self.data:
```
Comment (bdestombe) [inline flopy/modpath/mpsim.py]: This starts a write command per particle. This is relatively slow when you're about to simulate a lot of particles. Especially for slow hard disks. A suggestion would be to first store it in some sort of in-memory buffer. 

```python
s = ''
for p in self.data:
    s += '{:d} {:d} {:d} {:d} {:d} {:d}'.format(p.particleid, particlegroup, initialgrid, k0, p.i0, p.j0)
    fmtstr = ' {0} {0} {0} {0} '.format(float_format)
    s += fmtstr.format(p.xloc0, p.yloc0, p.zloc0, p.initialtime)
    s += '{}\n'.format(p.label.decode())

output.write(s)


```

File: flopy/modpath/mpsim.py [hunk 2]
```diff
@@ -280,3 +280,94 @@ def write_file(self):
...
+class StartingLocations():
+    """Class for working with MODPATH Starting Locations file for particles.
+
+    Parameters
+    ----------
+    model : Modpath object
+        The model object (of type :class:`flopy.modpath.mp.Modpath`) to which
+        this package will be added.
+    inputstyle : 1
+        Input style described in MODPATH6 manual (currently only input style 1 is supported)
```
Comment (bdestombe) [inline flopy/modpath/mpsim.py]: Maybe catch this with a NotImplementedError?

Comment (aleaf) [inline flopy/modpath/mpsim.py]: Bas, I made some changes to address both of these comments.

Comment (bdestombe) [inline flopy/modpath/mpsim.py]: Looks great, thanks! What a huge commit btw!

**Pull request merged**